app/src/main/java/app/MainActivity.kt

```diff
@@ -1,12 +1,114 @@
+    Returns null if an error occurred
+     */
+    fun getPrompts(): MutableList<Prompt>? {
+        val queue= LinkedBlockingQueue<MutableList<Prompt>?>()
+        thread {
+            try{
+            //StrictMode.setThreadPolicy(StrictMode.ThreadPolicy.Builder().detectNetwork().permitAll().build())
+            val yahoo = URL("https://api.jsonbin.io/v3/b/62ca4dbc5d53821c3097eaef/")
+            val inStream = BufferedReader(InputStreamReader(yahoo.openStream()))
+
+            var inputLine = inStream.readLine()
+            var data = "";
+
+            while (inputLine != null) {
+                println(inputLine)
+                data += inputLine;
+                inputLine = inStream.readLine();
+            }
+            inStream.close()
+
+            val prompts = mutableListOf<Prompt>()
+            //array way
+            /*val jRecords=JSONObject(data).get("record") as JSONArray;
+        println("RECORD");
+          println(jRecords.toString())
+        for (i in 0 until jRecords.length()){
+            val jPrompt=jRecords.get(i) as JSONObject
+            //println(jPrompt)
+            val promptText=jPrompt.get("PromptText").toString()
+            val energy=jPrompt.get("Energy").toString().toInt()
+            val money=jPrompt.get("Money").toString().toInt()
+            val status=jPrompt.get("Status").toString().toInt()
+            val left=jPrompt.get("Left").toString().toInt()
+            val right=jPrompt.get("Right").toString().toInt()
+            println("{" +
+                    "\n promptText: "+promptText +
+                    "\n energy: "+energy +
+                    "\n money: "+money +
+                    "\n status: "+status +
+                    "\n left: "+left +
+                    "\n right: "+right +
+                    "\n}")
+            prompts.add(Prompt(promptText,energy,money,status,left,right))
+        }*/
+
+            val jRecord = JSONObject(data).get("record") as JSONObject;
+            var i = 1;
+            while (true) {
+                if (!jRecord.has(i.toString()))
+                    break;
+
+                val jPrompt = jRecord.get(i.toString()) as JSONObject
+                println(jPrompt)
+                val promptText = jPrompt.get("PromptText").toString()
+                val energy = jPrompt.get("Energy").toString().toInt()
+                val money = jPrompt.get("Money").toString().toInt()
+                val status = jPrompt.get("Status").toString().toInt()
+                val left = jPrompt.get("Left").toString().toInt()
+                val right = jPrompt.get("Right").toString().toInt()
+                println("{" +
+                        "\n promptText: " + promptText +
+                        "\n energy: " + energy +
+                        "\n money: " + money +
+                        "\n status: " + status +
+                        "\n left: " + left +
+                        "\n right: " + right +
+                        "\n}")
```

Comment:
Use proper logging instead of println()
- Logging has the concept of logging levels.  They can be filtered and aggregated based on those levels.
- No one will see the output of println().